src/app/shared/navigation-focus/navigation-focus.ts

```diff
@@ -9,16 +11,32 @@ let lastTimeoutId = -1;
 export class NavigationFocus implements OnInit {
   @HostBinding('tabindex') role = '-1';
 
-  constructor(private el: ElementRef) {}
+  constructor(private el: ElementRef, private router: Router) {}
 
   ngOnInit() {
-    clearTimeout(lastTimeoutId);
-    // 100ms timeout is used to allow the page to settle before moving focus for screen readers.
-    lastTimeoutId = window.setTimeout(() => this.el.nativeElement.focus({preventScroll: true}),
-      100);
+    let previousUrl = window.location.hostname;
+    this.router.events
+      .pipe(filter((event: Event) => event instanceof NavigationEnd))
+      .subscribe(() => {
+        const afterUrl = window.location.hostname;
+        // only want to focus on the element if it the navigation was a soft nav,
+        // otherwise first focusable element should be the skip link
+        if (isSoftNav(previousUrl)) {
```

Comment:
Can you determine this from the `id` on the `RouterEvent` https://angular.io/api/router/RouterEvent? A hard nav would have an id of 0, right?

Reply:
I figured there had to be some way, but I had no idea what it was. Also, I would never guess that that's what 0 means from the docs

Reply:
Yea, it's not immediately apparent. Using the `id` won't work if the initial navigation results in a redirect from a guard calling `router.navigate` or returning `UrlTree`. The initial navigation would then be cancelled and the following would have `id` 1... Still trying to think of a more "Angular" way to do this that works in all cases. 
I'm not 100% confident that the current way would work in the redirect case either. @vxlydia - do you think you can you test this out?

Reply:
I tried checking if `id` is 0 and it seems so work. I'm not sure how to test for the redirect case, could you point me to an example?

Reply:
The only guard I could find was here: https://github.com/angular/material.angular.io/blob/master/src/app/pages/component-sidenav/component-sidenav-can-load-guard.ts and it does call `router.navigate`. So to test out this cancellation/redirect behavior, navigate to `localhost:4200/nonsenseroute`.